devito/profiler.py

```diff
@@ -39,50 +95,208 @@ def add_profiling(self, block, name, omp_flag=None):
                            "(double)(end_%(n)s.tv_sec-start_%(n)s.tv_sec)+" +
                            "(double)(end_%(n)s.tv_usec-start_%(n)s.tv_usec)" +
                            "/1000000") %
-                          {"sn": self.name, "n": name}),
+                          {"sn": self.t_name, "n": name}),
             ])
         ]
 
-        return init + block + end
+        return init + code + end
 
-    @property
-    def get_timings_class(self):
+    def get_oi_and_flops(self, name, code, size):
+        """Calculates the total operation intensity of the code provided.
+        If needed, lets the C code calculate it.
+
+        :param name: The name of the field to be populated
+        :param code: The code to be profiled.
+        """
+        if name not in self.oi:
+            self.oi[name] = 0
```

Comment:
Consider using a ``defaultdict(int)`` instead of a ``{}`` ?

``from collections import defaultdict``